atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1785,7 +1786,8 @@ protected LockToken acquireLocksForCommit() {
                 continue;
             }
             ConflictHandler conflictHandler = getConflictHandlerForTable(tableRef);
-            if (conflictHandler == ConflictHandler.RETRY_ON_WRITE_WRITE_CELL) {
+            if (conflictHandler == ConflictHandler.RETRY_ON_WRITE_WRITE_CELL
+                    || conflictHandler == ConflictHandler.SERIALIZABLE_CELL) {
```

Comment:
This is the heart of the change, and is correct.

nit: Maybe have helpers `mustCheckWriteWriteConflicts()` and `useCellLevelLocking()`?

Reply:
or even better, have ConflictHandler have that method on it!

Reply:
Yep, that would work - though there is an awkward `RETRY_ON_VALUE_CHANGED` that I think we'll still need to reference explicitly. It makes sense to add check w/w conflicts and lock levels to the enum, though I don't know what people feel in general about enum members that are equivalent in all aspects but unequal 🤷‍♂️